ios/brave-ios/Sources/Brave/Frontend/Browser/TabWKUIHandler.swift

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import BraveCore
+import Foundation
+import WebKit
+
+class TabWKUIHandler: NSObject, WKUIDelegate {
+  weak var tab: Tab?
+
+  init(tab: Tab) {
+    self.tab = tab
+  }
+
+  func webView(
+    _ webView: WKWebView,
+    createWebViewWith configuration: WKWebViewConfiguration,
+    for navigationAction: WKNavigationAction,
+    windowFeatures: WKWindowFeatures
+  ) -> WKWebView? {
+    guard let tab,
+      let childTab = tab.webDelegate?.tab(
+        tab,
+        createNewTabWithRequest: navigationAction.request,
+        configuration: configuration
+      )
+    else { return nil }
+    return childTab.webView
+  }
+
+  func webViewDidClose(_ webView: WKWebView) {
+    guard let tab else { return }
+    tab.webDelegate?.tabWebViewDidClose(tab)
+  }
+
+  func webView(
+    _ webView: WKWebView,
+    requestMediaCapturePermissionFor origin: WKSecurityOrigin,
+    initiatedByFrame frame: WKFrameInfo,
+    type: WKMediaCaptureType,
+    decisionHandler: @escaping (WKPermissionDecision) -> Void
+  ) {
+    guard let tab, let captureType = WebMediaCaptureType(type), let delegate = tab.webDelegate
+    else {
+      decisionHandler(.deny)
+      return
+    }
+
+    if frame.securityOrigin.protocol.isEmpty || frame.securityOrigin.host.isEmpty {
```

Comment:
I was confused whether we needed the `frame.securityOrigin` or the `origin`, but `frame.securityOrigin` is correct :)